### ISSUE TITLE
LC0028 gives false positives in pre-release v0.30.8

### ISSUE THREAD

**Issue by jwikman** [StefanMaron/BusinessCentral.LinterCop#368]
The LC0028 rule appears for all event subscriber, after downloaded latest pre-release, v0.30.8

As a repro, just add any event, like below, and LC0028 should be shown.
```
    [EventSubscriber(ObjectType::Table, Database::"VAT Registration No. Format", OnBeforeTest, '', false, false)]
    local procedure OnBeforeTest(VATRegNo: Text[20]; CountryCode: Code[10]; Number: Code[20]; TableID: Option; Check: Boolean; var IsHandled: Boolean);
    begin
    end;
```

**Comment by pri-kise** [comment]
I can confirm this.
Maybe this is related to AL Language Extensions.
I'm using the preview version there, too.

**Comment by jwikman** [comment]
I get this issue with AL Language Extension v12.3.900902

**Comment by pri-kise** [comment]
There has been some changes by @Arthurvdv  https://github.com/StefanMaron/BusinessCentral.LinterCop/commit/7ab27742afd59465a4a5e05adac612df1b39f2bd#diff-a0e92125998f6d044c26a7d889672ddfdc0b0984a7ecdf99079b38de0b3628f7

maybe those are related to this issue

**Comment by fvet** [comment]
Having the same warnings, running on

Name: AL Language extension for Microsoft Dynamics 365 Business Central
Id: ms-dynamics-smb.al
Description: Development tools for Dynamics 365 Business Central
Version: **13.0.903630**
Publisher: Microsoft
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=ms-dynamics-smb.al

**Comment by Arthurvdv** [comment]
The prerelease has been updated (same version number), where this issue should be resolved.

> maybe those are related to this issue
Spot on! The rule is now extended that it also verifies the second parameter. 
So the example below now also will be catched by the LC0028 rule.

```AL
[EventSubscriber(ObjectType::Page, Page::"Sales Order", OnAfterActionEvent, 'Archive Document', false, false)]
local procedure MyProcedure()
begin

end;

```

**Comment by jwikman** [comment]
Now I get this exception instead:
```
Analyzer 'BusinessCentral.LinterCop.Design.Rule0028CodeNavigabilityOnEventSubscribers' threw an exception of type 'System.NullReferenceException' with message 'System.NullReferenceException: Object reference not set to an instance of an object.
   at BusinessCentral.LinterCop.Design.Rule0028CodeNavigabilityOnEventSubscribers.CodeNavigabilityOnEventSubscribers(CodeBlockAnalysisContext context)
   at Microsoft.Dynamics.Nav.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c__DisplayClass55_2`2.<ExecuteBlockActions>b__1() in D:\a\_work\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\DiagnosticAnalyzer\AnalyzerExecutor.cs:line 922
   at Microsoft.Dynamics.Nav.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock(DiagnosticAnalyzer analyzer, Action analyze, Nullable`1 info) in D:\a\_work\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\DiagnosticAnalyzer\AnalyzerExecutor.cs:line 1102'
```

**Comment by tscottjendev** [comment]
Getting the same error, 
[{
	"resource": "/c:/Repos/Jendev/JI-InternalSystem-NEW/DataCleanse/app.json",
	"owner": "_generated_diagnostic_collection_name_#2",
	"code": "AD0001",
	"severity": 4,
	"message": "Analyzer 'BusinessCentral.LinterCop.Design.Rule0028CodeNavigabilityOnEventSubscribers' threw an exception of type 'System.NullReferenceException' with message 'System.NullReferenceException: Object reference not set to an instance of an object.\r\n   at BusinessCentral.LinterCop.Design.Rule0028CodeNavigabilityOnEventSubscribers.CodeNavigabilityOnEventSubscribers(CodeBlockAnalysisContext context)\r\n   at Microsoft.Dynamics.Nav.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c__DisplayClass55_2`2.<ExecuteBlockActions>b__1() in D:\\a\\_work\\1\\s\\source\\Prod\\Microsoft.Dynamics.Nav.CodeAnalysis\\DiagnosticAnalyzer\\AnalyzerExecutor.cs:line 922\r\n   at Microsoft.Dynamics.Nav.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock(DiagnosticAnalyzer analyzer, Action analyze, Nullable`1 info) in D:\\a\\_work\\1\\s\\source\\Prod\\Microsoft.Dynamics.Nav.CodeAnalysis\\DiagnosticAnalyzer\\AnalyzerExecutor.cs:line 1102'",
	"source": "AL",
	"startLineNumber": 1,
	"startColumn": 1,
	"endLineNumber": 1,
	"endColumn": 1
}]

**Comment by jwikman** [comment]
I could not repro either in a small app I tested in.

Got to leave now, can check back later tonight or tomorrow if I can do a clean repro.

**Comment by tscottjendev** [comment]
Seems to be maybe related to a multi-root workspace.  I open my normal workspace and I get a lot of these errors in 4 of 10 app folders.  One is very small so I open its folder (and edit its app.json) by itself and I don't get any errors.  Re-open the workspace, that same app reports 11 errors and there are **no** event subscribers in it.

If I open another app (edit the app.json) in the workspace, it shows 46 errors and has **no** subscribers but a few publishers.

**Comment by tscottjendev** [comment]
What's even weirder is that it is not always the same apps reporting the errors.  I reopened my workspace again and now 4 different apps are reporting errors and neither of the ones from the previous comment are in the 4.

**Comment by tscottjendev** [comment]
And the errors for an app go away when I double click on the entry in the "Problems" window, until I reload the workspace again.

**Comment by Arthurvdv** [comment]
> Seems to be maybe related to a multi-root workspace.

I've setup an multi-root workspace, unfortunately no succes to reproduce this on my side.
..but this gave me an idea where I maybe could resolve this (not sure though, I'm flying blind here).

I've updated the existing prelease version [v0.30.8](https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/tag/v0.30.8), where I hope this will resolve this issue?

**Comment by jwikman** [comment]
Nope, sorry. Still the very same error.

And it is not related to the multi-root workspace. I first thought so too, but then I realized that we've configured the Code Analyzers in the workspace. When I activated the code analysis with LC for the app, I got the very same error again.

I tried to set up and debug the LC code from Visual Studio, but I did not get the debugger to work... Are there any documentation on how to set up LC for debugging? If so, please let me know where to find info on how to get started and I will be able to provide better feedback! (and maybe some PRs  later on 🙂)
